service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -190,8 +223,267 @@ public void prepareCustomDirections(Map<String, Object> data) {
+
+        cmoTemplateData.put("representatives",
+            getRepresentatives(caseData.getRespondents1(), applicantName, caseData.getSolicitor()));
+
+        // Populate with the next hearing booking, currently not captured
+        cmoTemplateData.putAll(commonCaseDataExtractionService.getHearingBookingData(null));
+
+        HearingBooking hearingBooking = getHearingBooking(caseData.getHearingDetails(), hearingDateList);
+        JudgeAndLegalAdvisor judgeAndLegalAdvisor = hearingBooking.getJudgeAndLegalAdvisor();
+        cmoTemplateData.put("judgeTitleAndName", defaultIfBlank(formatJudgeTitleAndName(judgeAndLegalAdvisor),
+            EMPTY_PLACEHOLDER));
+        cmoTemplateData.put("legalAdvisorName", defaultIfBlank(getLegalAdvisorName(
+            judgeAndLegalAdvisor), EMPTY_PLACEHOLDER));
+
+        cmoTemplateData.putAll(getGroupedCMODirections(caseManagementOrder));
+
+        cmoTemplateData.put("draftbackground", String.format("image:base64:%1$s",
+            docmosisDocumentGeneratorService.generateDraftWatermarkEncodedString()));
+
+        List<Map<String, String>> recitals = buildRecitals(caseManagementOrder.getRecitals());
+        cmoTemplateData.put("recitals", recitals);
+        cmoTemplateData.put("recitalsProvided", isNotEmpty(recitals));
+
+        cmoTemplateData.putAll(getSchedule(caseManagementOrder));
+
+        //defaulting as 1 as we currently do not have impl for multiple CMos
+        cmoTemplateData.put("caseManagementNumber", 1);
+
+        return cmoTemplateData.build();
+    }
+
+    private List<Map<String, Object>> getRepresentatives(List<Element<Respondent>> respondents1,
+                                                         String applicantName,
+                                                         Solicitor solicitor) {
+        List<Map<String, Object>> representatives = new ArrayList<>();
+
+        representatives.add(getApplicantDetails(applicantName, solicitor));
+
+        if (!isEmpty(respondents1)) {
+            respondents1.stream()
+                .map(Element::getValue)
+                .forEach(respondent -> representatives.add(getRepresentative(respondent)));
+        }
+
+        return representatives;
+    }
+
+    private Map<String, Object> getRepresentative(Respondent respondent) {
+        return ImmutableMap.of(
+            "respondentName", respondent.getParty().getFullName(),
+            "representativeName", EMPTY_PLACEHOLDER,
+            "representativeEmail", EMPTY_PLACEHOLDER,
+            "representativePhoneNumber", EMPTY_PLACEHOLDER
+        );
+    }
+
+    private Map<String, Object> getSchedule(CaseManagementOrder caseManagementOrder) {
+        final Schedule schedule = caseManagementOrder.getSchedule();
+        Map<String, Object> scheduleMap = mapper.convertValue(schedule, new TypeReference<>() {});
+        Map<String, Object> result;
+
+        if ((schedule == null) || schedule.getIncludeSchedule().equals("No")) {
```

Comment:
Would probably eextract to if(isScheduleIncluded(schedule)) would read a lot more nicely, or do some optional mapping.

Reply:
Yeah will extract it to a method